syntax = "proto3";

option go_package = "pb";

// import "common.proto";
// import "entity_search_highlight.proto";

package entity;

message Empty {}

message Pager {
    int32 page = 1;
    int32 offset = 2;
}

message ArticleID {
    string id = 1;
}

// message ArticleIDList {
//     repeated string ids = 1;
// }

// message ArticleIDWithPager {
//     string id = 1;
//     // common.Pager pager = 2;
// }

message Article {
    string id = 1;
    string title = 2;
    string description = 3;
    string userID = 4;
    string createdAt = 5;
    string updatedAt = 6;
    // string thumbnailURL = 6;
    // repeated entity.SearchHighlight highlights = 6;
}

message ArticleList {
    repeated Article items = 1;
}

message CreateArticleRequest {
    string title = 1;
    string description = 2;
    string userID = 3;
    // string thumbnailURL = 3;
    // string createdAt = 3;
}

message UpdateArticleRequest {
    string id = 1;
    string title = 2;
    string description = 3;
    // string thumbnailURL = 4;
}
